From 338b86c4f0c2cdc4241dbf5cda913f0184afc105 Mon Sep 17 00:00:00 2001
From: Huzaifa Sidhpurwala <huzaifas@redhat.com>
Date: Tue, 26 Apr 2011 17:15:56 +0000
Subject: dialog: Fix crash in user switcher code

The user switch button currently causes the lock dialog to crash
because of an inverted conditional in the error checking code.

This commit addresses the crash by performing the proper check
in the conditional.

https://bugzilla.gnome.org/show_bug.cgi?id=648234
---
diff --git a/src/gs-lock-plug.c b/src/gs-lock-plug.c
index 67ab36a..6706fde 100644
--- a/src/gs-lock-plug.c
+++ b/src/gs-lock-plug.c
@@ -167,7 +167,7 @@ do_user_switch (GSLockPlug *plug)
         g_object_unref (context);
         g_object_unref (app);
 
-        if (!error) {
+        if (error != NULL) {
                 gs_debug ("Unable to start GDM greeter: %s", error->message);
                 g_error_free (error);
         }
--
cgit v0.9
